server/validation/validation.go

```diff
@@ -43,5 +43,27 @@ func ValidateObjectsRelations(ctx context.Context, backend storage.TypeDefinitio
 	if !ok {
 		return nil, &tuple.RelationNotFoundError{Relation: t.GetRelation(), TypeName: ns.GetType(), TupleKey: t}
 	}
+
+	// if the user is a userset, validate the type and relation on it as well
+	user := t.GetUser()
```

Comment:
Should this part be in the `ValidateTuple` function as this function is called `ValidateObjectsRelations`?

Also, what if the user is just an object, like `folder:X`, and not an ObjectRelation? Should we be validating that the type `folder` exists? Or is this done elsewhere?

Reply:
> Should this part be in the ValidateTuple function as this function is called ValidateObjectsRelations?

Good point! Yes! `ValidateObjectsRelations` is being called in Expand, which doesn't take a `user` param at all, so it's pointless to call that code. I moved it to its own function now.

> what if the user is just an object, like folder:X, and not an ObjectRelation? Should we be validating that the type folder exists? Or is this done elsewhere?

If the model is 1.1, this is done in https://github.com/openfga/openfga/blob/776e80505e8d184b2286acc8268d8d74f36a9984/server/commands/write.go#L161

Reply:
We really need https://github.com/openfga/openfga/pull/341. Now i don't know whether https://github.com/openfga/openfga/issues/189 applies to 1.0 models, 1.1 models, or both...

Reply:
Ah, yes, good point. I forgot that we allow user_ids to be anything in the 1.0 world. oh my.